Can not build fusio-log without default features

```
❯ cargo build --no-default-features
   Compiling thiserror-impl v2.0.12
   Compiling fusio v0.3.7 (/Users/gwo/Idea/fusio/fusio)
   Compiling fusio-dispatch v0.3.7 (/Users/gwo/Idea/fusio/fusio-dispatch)
   Compiling thiserror v2.0.12
   Compiling fusio-log v0.3.7 (/Users/gwo/Idea/fusio/fusio-log)
error[E0599]: no variant or associated item named `S3Error` found for enum `fusio::Error` in the current scope
  --> fusio-log/src/error.rs:28:27
   |
28 |             fusio::Error::S3Error(_) => LogError::S3Error(err),
   |                           ^^^^^^^ variant or associated item not found in `Error`

error[E0599]: no variant or associated item named `Local` found for enum `FsOptions` in the current scope
  --> fusio-log/src/option.rs:24:35
   |
24 |             fs_option: FsOptions::Local,
   |                                   ^^^^^ variant or associated item not found in `FsOptions`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `fusio-log` (lib) due to 2 previous errors
```

--- a/fusio-log/Cargo.toml
+++ b/fusio-log/Cargo.toml
@@ -10,10 +10,7 @@ version.workspace = true
 aws = ["fusio-dispatch/aws", "fusio/aws"]
 bytes = ["dep:bytes"]
 default = ["aws", "bytes", "tokio", "tokio-http"]
-tokio = [
-    "fusio-dispatch/tokio",
-    "fusio/tokio",
-]
+tokio = ["fusio-dispatch/tokio", "fusio/tokio"]
 tokio-http = ["fusio/tokio-http"]
 web = ["fusio-dispatch/opfs", "fusio/opfs"]
 web-http = ["fusio/wasm-http"]
@@ -24,9 +21,9 @@ crc32fast = "1"
 fusio = { version = "0.3.7", path = "../fusio", features = [
     "dyn",
     "fs",
-    "bytes"
+    "bytes",
 ] }
-fusio-dispatch = {  version = "0.3.7", path = "../fusio-dispatch" }
+fusio-dispatch = { version = "0.3.7", path = "../fusio-dispatch" }
 futures-core = "0.3"
 futures-util = "0.3"
 thiserror = "2.0.3"

--- a/fusio-log/src/error.rs
+++ b/fusio-log/src/error.rs
@@ -2,9 +2,11 @@ use thiserror::Error;
 
 #[derive(Debug, Error)]
 #[error("log error")]
+#[non_exhaustive]
 pub enum LogError {
     #[error("IO error: {0}")]
     IO(#[from] std::io::Error),
+    #[cfg(feature = "aws")]
     #[error("S3 error: {0}")]
     S3Error(fusio::Error),
     #[error("encode error: {message}")]
@@ -25,6 +27,7 @@ impl From<fusio::Error> for LogError {
     fn from(err: fusio::Error) -> Self {
         match err {
             fusio::Error::Io(error) => LogError::IO(error),
+            #[cfg(feature = "aws")]
             fusio::Error::S3Error(_) => LogError::S3Error(err),
             err => LogError::Other(Box::new(err)),
         }

--- a/fusio-log/src/lib.rs
+++ b/fusio-log/src/lib.rs
@@ -78,6 +78,7 @@ impl<T> Logger<T>
 where
     T: Encode,
 {
+    /// Write a batch of log entries to the log file.
     pub async fn write_batch<'r>(
         &mut self,
         data: impl ExactSizeIterator<Item = &'r T>,
@@ -98,6 +99,8 @@ where
         Ok(())
     }
 
+    /// Write a single log entry to the log file. This method has the same behavior as `write_batch`
+    /// but for a single entry.
     pub async fn write(&mut self, data: &T) -> Result<(), LogError> {
         let mut writer = HashWriter::new(&mut self.buf_writer);
 
@@ -111,11 +114,14 @@ where
         Ok(())
     }
 
+    /// Flush the data to the log file.
     pub async fn flush(&mut self) -> Result<(), LogError> {
         self.buf_writer.flush().await?;
         Ok(())
     }
 
+    /// Close the log file. This will flush the data to the log file and close it.
+    /// It is not guaranteed that the log file can be operated after closing.
     pub async fn close(&mut self) -> Result<(), LogError> {
         self.buf_writer.close().await?;
         Ok(())
@@ -405,6 +411,46 @@ mod tests {
             let mut stream = Options::new(path).recover::<TestStruct>().await.unwrap();
             let res = stream.try_next().await.unwrap();
             assert!(res.is_none());
+        }
+    }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn test_disable_buffer() {
+        let temp_dir = TempDir::new().unwrap();
+        let path = Path::from_filesystem_path(temp_dir.path())
+            .unwrap()
+            .child("disable_buf");
+
+        {
+            let mut logger = Options::new(path.clone())
+                .disable_buf()
+                .build::<u8>()
+                .await
+                .unwrap();
+            logger.write(&1).await.unwrap();
+            logger.write_batch([2, 3, 4].iter()).await.unwrap();
+            logger
+                .write_batch([2, 3, 4, 5, 1, 255].iter())
+                .await
+                .unwrap();
+            logger.flush().await.unwrap();
+            logger.close().await.unwrap();
+        }
+        {
+            let expected = [vec![1], vec![2, 3, 4], vec![2, 3, 4, 5, 1, 255]];
+            let stream = Options::new(path)
+                .recover::<u8>()
+                .await
+                .unwrap()
+                .into_stream();
+            pin!(stream);
+            let mut i = 0;
+            while let Some(res) = stream.next().await {
+                assert!(res.is_ok());
+                assert_eq!(&expected[i], &res.unwrap());
+                i += 1;
+            }
+            assert_eq!(i, expected.len())
         }
     }
 }

--- a/fusio-log/src/option.rs
+++ b/fusio-log/src/option.rs
@@ -8,6 +8,15 @@ use crate::{error::LogError, Decode, Encode, Logger};
 
 pub(crate) const DEFAULT_BUF_SIZE: usize = 4 * 1024;
 
+#[cfg(not(any(feature = "tokio", feature = "web", feature = "aws")))]
+compile_error!("one of these features must be enabled: tokio, web, aws");
+
+#[cfg(all(
+    feature = "aws",
+    not(any(feature = "tokio-http", feature = "web-http"))
+))]
+compile_error!("aws feature must be used with tokio-http or web-http feature");
+
 #[derive(Clone)]
 pub struct Options {
     pub(crate) path: fusio::path::Path,
@@ -17,6 +26,8 @@ pub struct Options {
 }
 
 impl Options {
+    /// Create a new log options.
+    #[cfg(any(feature = "tokio", feature = "web"))]
     pub fn new(path: fusio::path::Path) -> Self {
         Self {
             path,
@@ -26,6 +37,20 @@ impl Options {
         }
     }
 
+    /// Create a new log options with [`FsOptions`].
+    ///
+    /// See [`FsOptions`] for more details.
+    pub fn with_fs_options(path: fusio::path::Path, fs_option: FsOptions) -> Self {
+        Self {
+            path,
+            buf_size: DEFAULT_BUF_SIZE,
+            fs_option,
+            truncate: false,
+        }
+    }
+
+    /// Disable buffer for the log. It is recommended to keep the buffer enabled and use
+    /// [`Logger::flush`] to flush the data to the log file.
     pub fn disable_buf(self) -> Self {
         Self {
             buf_size: 0,
@@ -33,18 +58,24 @@ impl Options {
         }
     }
 
+    /// Set the buffer size for the log. Default is 4K.
     pub fn buf_size(self, buf_size: usize) -> Self {
         Self { buf_size, ..self }
     }
 
+    /// Set the filesystem options for the log.
+    ///
+    /// See [`FsOptions`] for more details.
     pub fn fs(self, fs_option: FsOptions) -> Self {
         Self { fs_option, ..self }
     }
 
+    /// Open log file with truncate option.
     pub fn truncate(self, truncate: bool) -> Self {
         Self { truncate, ..self }
     }
 
+    /// Open the log file. Return error if open file failed.
     pub async fn build<T>(self) -> Result<Logger<T>, LogError>
     where
         T: Encode,
@@ -53,6 +84,7 @@ impl Options {
         Ok(logger)
     }
 
+    /// Open the log with the given [`DynFs`]. Return error if open file failed.
     pub async fn build_with_fs<T>(self, fs: Arc<dyn DynFs>) -> Result<Logger<T>, LogError>
     where
         T: Encode,
@@ -61,6 +93,7 @@ impl Options {
         Ok(logger)
     }
 
+    /// Recover the log from existing log file. Return a stream of log entries.
     pub async fn recover<T>(
         self,
     ) -> Result<impl TryStream<Ok = Vec<T>, Error = LogError> + Unpin, LogError>

